### PR TITLE
dotCMS/core#26666 [UI] When the BlockEditor lose the focus, and get it again the generated content is duplicated 

### LINKED ISSUE #26666: [UI] When the BlockEditor lose the focus, and get it again the generated content is duplicated
### User Story
When the BlockEditor lose the focus, and get it again the generated content is duplicated

Screenshot 2023-11-13 at 1 53 17 PM

### Acceptance Criteria
The content should not be duplicated. 

### dotCMS Version
master

### Proposed Objective
_to_define_

### Proposed Priority
_to_define_

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.
_to_define_

### Assumptions & Initiation Needs
_to_define_

### Quality Assurance Notes & Workarounds
_to_define_

### Sub-Tasks & Estimates

SP: 3

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* Content is now not duplicated when the user click in the editor + accept the AI content. 
* Remove trust in editor `selection` to show / replace / remove de AI content node. Instead a new functions that identify the position of a specific node type to eventually be manipulated. 
* The `insertAINode` now identify if there is a current node in the document, if so, replace the content on it otherwise create a new node and add it to the editor. 
* Handle the click event manually, show the prompt stays in the editor while waiting the response from the AI services. 
*   create functions `findNodeByType ` and  `replaceNodeWithContent ` to make easier node manipulation with tiptap.

### CODE REVIEW

File: core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts [hunk 1]
```diff
@@ -94,23 +109,59 @@ export class AIContentPromptView {
...
+                this.componentStore.setExit(false);
+            });
+
+        /**
+         * Subscription to delete AI_CONTENT node.
+         * Fired from the AI Content Actions plugin.
+         */
+        this.componentStore.deleteContent$
+            .pipe(
+                skip(1),
```
Comment (zJaaal) [inline core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts:146]: Why do we need to skip one?

**Review by zJaaal — APPROVED** [review]

File: core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts (hunk 1 above)
Comment (hmoreras) [reply core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts:146]: to skip the state initialization.

**Review by oidacra — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]

File: core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/store/ai-content-prompt.store.ts [hunk 2]
```diff
@@ -13,28 +13,49 @@ export interface AiContentPromptState {
...
     constructor(private dotAiService: DotAiService) {
-        super({ prompt: '', loading: false, content: '', open: false, acceptContent: false });
+        super({
+            prompt: '',
+            loading: false,
+            content: '',
+            open: false,
+            acceptContent: false,
+            deleteContent: false,
+            exit: false
```
Comment (fmontes) [inline core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/store/ai-content-prompt.store.ts:32]: loading, open and exit... is there any way more that one can happen at the same time? maybe we do like: `state: 'idle' | 'loading' | 'open'`?

**Review by fmontes — APPROVED** [review]

File: core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/store/ai-content-prompt.store.ts (hunk 2 above)
Comment (hmoreras) [reply core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/store/ai-content-prompt.store.ts:32]: agree moved to `status`

**Pull request merged**